src/backends/plonky2/circuits/mainpod.rs

```diff
@@ -137,91 +137,211 @@ impl OperationVerifyGate {
     ) -> Result<OperationVerifyTarget> {
         let _true = builder._true();
         let _false = builder._false();
-        let one = builder.constant(F::ONE);
 
         // Verify that the operation `op` correctly generates the statement `st`.  The operation
         // can reference any of the `prev_statements`.
+        // TODO: Clean this up.
+        let resolved_op_args = if prev_statements.len() == 0 {
+            vec![]
+        } else {
+            op.args
+                .iter()
+                .flatten()
+                .map(|&i| {
+                    StatementTarget::from_flattened(
+                        builder.matrix_row_ref(
```

Comment:
Idea: what if we define a trait to flatten and unflatten:
```rust
trait Flatable {
  fn flaten(&self) -> Vec<Target>;
  fn from_flattened(vs: &[Target]) -> Self;
}
```

And then define some generic methods lile:
```
fn select<T: Flatable>(b: BoolTarget, x: T, y: T) -> T;
fn vec_ref<T: Flatable>(i: usize, vs: &[T]) -> T;
```
These methods would be implemented to flatten the types, apply the logic, and reconstruct the `T`